rubyUtils.gradle

```diff
@@ -0,0 +1,244 @@
+    }
+}
+
+import de.undercouch.gradle.tasks.download.Download
+import de.undercouch.gradle.tasks.download.Verify
+import org.yaml.snakeyaml.Yaml
+
+
+import org.jruby.Ruby
+import org.jruby.embed.PathType
+import org.jruby.embed.ScriptingContainer
+
+import java.nio.file.Files
+import java.nio.file.Paths
+
+ext {
+    bundle = this.&bundle
+    bundleWithEnv = this.&bundleWithEnv
+    gem = this.&gem
+    buildGem = this.&buildGem
+    rake = this.&rake
+    versionMap = new HashMap()
+}
+
+/**
+ * Executes a bundler bin script with given parameters.
+ * @param projectDir Gradle projectDir
+ * @param buildDir Gradle buildDir
+ * @param pwd Current worker directory to execute in
+ * @param bundleBin Bundler Bin Script
+ * @param args CLI Args to Use with Bundler
+ */
+void bundle(File projectDir, File buildDir, String pwd, String bundleBin, Iterable<String> args) {
+    bundleWithEnv(projectDir, buildDir, pwd, bundleBin, args, Collections.emptyMap())
+}
+
+/**
+ * Executes a bundler bin script with given parameters.
+ * @param projectDir Gradle projectDir
+ * @param buildDir Gradle buildDir
+ * @param pwd Current worker directory to execute in
+ * @param bundleBin Bundler Bin Script
+ * @param args CLI Args to Use with Bundler
+ * @param env Environment Variables to Set
+ */
+void bundleWithEnv(File projectDir, File buildDir, String pwd, String bundleBin, Iterable<String> args, Map<String, String> env) {
+    executeJruby projectDir, buildDir, { ScriptingContainer jruby ->
+        jruby.environment.putAll(env)
+        jruby.currentDirectory = pwd
+        jruby.argv = args.toList().toArray()
+        jruby.runScriptlet(PathType.ABSOLUTE, bundleBin)
+    }
+}
+
+/**
+ * Installs a Gem with the given version to the given path.
+ * @param projectDir Gradle projectDir
+ * @param buildDir Gradle buildDir
+ * @param gem Gem Name
+ * @param version Version to Install
+ * @param path Path to Install to
+ */
+void gem(File projectDir, File buildDir, String gem, String version, String path) {
+    executeJruby projectDir, buildDir, { ScriptingContainer jruby ->
+        jruby.currentDirectory = projectDir
+        jruby.runScriptlet(
+                "require 'rubygems/commands/install_command'\n" +
+                        "cmd = Gem::Commands::InstallCommand.new\n" +
+                        "cmd.handle_options [\"--no-ri\", \"--no-rdoc\", '${gem}', '-v', '${version}', '-i', '${path}']\n" +
+                        "begin \n" +
+                        "  cmd.execute\n" +
+                        "rescue Gem::SystemExitException => e\n" +
+                        "  raise e unless e.exit_code == 0\n" +
+                        "end"
+        )
+    }
+}
+
+void buildGem(File projectDir, File buildDir, String gemspec) {
+    executeJruby projectDir, buildDir, { ScriptingContainer jruby ->
+        jruby.currentDirectory = projectDir
+        jruby.runScriptlet(
+                "require 'rubygems/commands/build_command'\n" +
```

Comment:
Not exactly related to your PR since you're just moving/namespacing these bits, but I _think_ that we can use groovy's triple-quote blocking combined with its `String#stripIndent()` to eliminate all of the quoting and newline escaping, which makes this a lot easier to read:

~~~ groovy
        jruby.runScriptlet("""\
                           require 'rubygems/commands/build_command'
                           cmd = Gem::Commands::BuildCommand.new
                           cmd.handle_options ['${gemspec}']
                           begin
                             cmd.execute
                           rescue Gem::SystemExitException => e
                             raise e unless e.exit_code == 0
                           end
                           """.stripIndent())
~~~

Reply:
That's definitely better. I wish Java would approve one of their multi-line string proposals.